src/anomalib/pipelines/components/base/pipeline.py

```diff
@@ -0,0 +1,74 @@
+"""Base class for orchestrator."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import warnings
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+from jsonargparse import ActionConfigFile, ArgumentParser, Namespace
+from rich import print, traceback
+
+traceback.install()
+log_file = "runs/pipeline.log"
+Path(log_file).parent.mkdir(exist_ok=True, parents=True)
+logger_file_handler = logging.FileHandler(log_file)
+logging.getLogger().addHandler(logger_file_handler)
+logging.getLogger().setLevel(logging.DEBUG)
+warnings.filterwarnings("ignore")
+for logger_name in ["lightning.pytorch", "lightning.fabric", "torchmetrics", "os"]:
+    logging.getLogger(logger_name).handlers = [logger_file_handler]
+format_string = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+logging.basicConfig(format=format_string, level=logging.DEBUG)
```

Comment:
Would it be possible to wrap this in a function, like `setup_logging` or something similar. It looks a bit messy this way